source/opt/optimizer.cpp

```diff
@@ -103,7 +104,8 @@ Optimizer& Optimizer::RegisterLegalizationPasses() {
           .RegisterPass(CreateLocalMultiStoreElimPass())
           // Copy propagate members.  Cleans up code sequences generated by
           // scalar replacement.
-          .RegisterPass(CreateInsertExtractElimPass())
+          .RegisterPass(CreateSimplificationPass())
+          //.RegisterPass(CreateInsertExtractElimPass())
```

Comment:
I left is because I was playing around with it. I'll remove the old code.

For -O, I want to add more rules and then replace InsertExtractElimPass.  There are still somethings the folder cannot do that the other pass does.

For -Os, I will make @greg-lunarg aware of what is being done, but let him decide what he wants to do with that set of optimizations.